### PR TITLE
feat:  add GUI parameter to select date formatter for the Excel importer

### PR DESCRIPTION
**Pull request by emanuelaepure10**
Add a combo box with different patterns for the date formatter. This pattern is then used when importing cells of type Date.

ING-4165

### CODE REVIEW

File: io/plugins/eu.esdihumboldt.hale.io.csv.ui/src/eu/esdihumboldt/hale/io/csv/ui/TypeSelectionPage.java [hunk 1]
```diff
@@ -122,16 +129,36 @@ public void selectionChanged(SelectionChangedEvent event) {
 		skipNlinesSpinner.setIncrement(1);
 		skipNlinesSpinner.setPageIncrement(10);
 
+//		create date formatter combo
+		Label dateFormatterLabel = new Label(page, SWT.NONE);
+		dateFormatterLabel.setText("Date formatter:");
```
Comment (florianesser) [inline io/plugins/eu.esdihumboldt.hale.io.csv.ui/src/eu/esdihumboldt/hale/io/csv/ui/TypeSelectionPage.java:134]: I'd suggest to put a more descriptive label here, e.g. `Format for imported date values`.

File: io/plugins/eu.esdihumboldt.hale.io.xls/src/eu/esdihumboldt/hale/io/xls/reader/ReaderSettings.java [hunk 2]
```diff
@@ -201,7 +201,8 @@ else if (skipType) {
 		}
 
 		// read dateFormat
-		String dateFormatString = reader.getParameter(PARAMETER_DATE_FORMAT).as(String.class);
+		String dateFormatString = reader.getParameter(CommonSchemaConstants.PARAM_DATE_FORMATTER)
```
Comment (florianesser) [inline io/plugins/eu.esdihumboldt.hale.io.xls/src/eu/esdihumboldt/hale/io/xls/reader/ReaderSettings.java:204]: Shouldn't this rather set [this parameter](https://github.com/emanuelaepure10/hale/blob/8deb946f73d1727e46ab7eda35835ec92866b4e5/io/plugins/eu.esdihumboldt.hale.io.xls/src/eu/esdihumboldt/hale/io/xls/reader/ReaderSettings.java#L60) that was introduced as part of [this PR](https://github.com/halestudio/hale/pull/1091)?

**Review by florianesser — CHANGES_REQUESTED** [review]

File: io/plugins/eu.esdihumboldt.hale.io.xls/src/eu/esdihumboldt/hale/io/xls/reader/ReaderSettings.java (hunk 2 above)
Comment (emanuelaepure10) [reply io/plugins/eu.esdihumboldt.hale.io.xls/src/eu/esdihumboldt/hale/io/xls/reader/ReaderSettings.java:204]: YES! Thanks

**Review by florianesser — APPROVED** [review]
Generally LGTM, can be merged IMO when the two remaining comments are addressed.

**Pull request merged**